### PR TITLE
Feature 20 login page

### PR DESCRIPTION
**Pull request by dimitrov93**

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -1,73 +1,92 @@
...
     component: Products,
-    meta: { title: "Products page" },
+    meta: { title: "Products page", requiresAuth: true },
   },
+
+  {
+    path: "/logout",
+    name: "Logout",
+    component: Products,
+    meta: { title: "Logout page", requiresAuth: true },
```
Comment (VladoKat) [inline src/router/index.js:86]: Do we need this? What are we showing on this page? It's a bit strange that we navigate here on a logout and requiresAuth is true. I may be misunderstanding this

File: src/utils/validation-rules.js [hunk 2]
```diff
@@ -18,3 +18,23 @@ export function useTextAreaFieldRules() {
...
+}
+
+export function usePasswordFieldRules() {
+  return [
+    (v) => !!v || "Password field is required",
+    (v) =>
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@#$%^&+=!])[A-Za-z\d@#$%^&+=!]{6,}$/.test(
+        v
+      ) || "Password must meet complexity requirements",
+    (v) => v.length >= 6 || "Password must be at least 6 characters long",
```
Comment (VladoKat) [inline src/utils/validation-rules.js:37]: These requirements would be needed if we wanted to create a new password. In Login form the user would be inputting an already existing password, so we don't need to do the complexity validation here.

**Review by VladoKat — CHANGES_REQUESTED** [review]
So far, these are the functional changes I've seen. I noticed there are console.logs. I haven't commented on those as I understand they are work in progress but they should be removed in the end.

File: src/utils/validation-rules.js (hunk 2 above)
Comment (dimitrov93) [reply src/utils/validation-rules.js:37]: fixed

File: src/router/index.js (hunk 1 above)
Comment (dimitrov93) [reply src/router/index.js:86]: fixed now but i faced issue with router.push('/login') after clicking logout on SM devices, turns out it goes to 404 for some reason, so i removed the router.push and added redirect in routes -   { path: "/logout", redirect: "/login" }. Now it works fine

File: src/store/modules/auth.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    clearUser(state) {
+      state.user = null
+      state.token = null
+    },
+  },
+  actions: {
+    async login({commit}, payload) {
+      const user = await postUserLogin(payload)
+      commit('setUser', user)
+      localStorage.setItem("userAuth", JSON.stringify(user.user));
```
Comment (VladoKat) [inline src/store/modules/auth.js:23]: didn't we say we are going to use vuex instead of the localStorage?

File: src/store/modules/auth.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      return user.user
+    },
+    logout({commit}) {
+      localStorage.removeItem("userAuth");
```
Comment (VladoKat) [inline src/store/modules/auth.js:27]: Same here

**Review by VladoKat — CHANGES_REQUESTED** [review]
Apart from directly calling the localStorage, the other looks good

File: src/store/modules/auth.js (hunk 4 above)
Comment (dimitrov93) [reply src/store/modules/auth.js]: done

File: src/store/modules/auth.js (hunk 3 above)
Comment (dimitrov93) [reply src/store/modules/auth.js]: done

**Review by VladoKat — APPROVED** [review]

**Pull request merged**